src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +327,32 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module,
+      ProjectFilesystem projectFilesystem,
+      Path extraCompileOutputRootPath) {
+    Path extraModuleRelativePath = extraCompileOutputRootPath.resolve(module.getModuleBasePath());
+    if (!projectFilesystem.exists(extraModuleRelativePath)) {
+      try {
+        projectFilesystem.mkdirs(extraModuleRelativePath);
```

Comment:
> When generating a real project (I used uber/okbuck sample app), without this logic, I get the exception below. It looks like the FakeProjectFilesystem used in the test code does not throw the same exception since it's .exists() method only checks whether its a file or directory and not the presence of the actual file on disk

Let's fix `IjProjectTemplateDataPreparer.createExcludes` instead of hacking in this class:

```
  public ImmutableCollection<IjFolder> createExcludes(final IjModule module) throws IOException {
    final Path moduleBasePath = module.getModuleBasePath();
    if (!projectFilesystem.exists(moduleBasePath)) {
      return ImmutableList.of();
    }
    final ImmutableList.Builder<IjFolder> excludesBuilder = ImmutableList.builder();
    projectFilesystem.walkRelativeFileTree(
```